diff -urN Delphes_V_2.0.0/src/SmearUtil.cc Delphes_V_2.0.0_ptiso/src/SmearUtil.cc
--- Delphes_V_2.0.0/src/SmearUtil.cc	2012-02-01 21:30:46.000000000 +0900
+++ Delphes_V_2.0.0_ptiso/src/SmearUtil.cc	2012-04-16 18:14:48.188280238 +0900
@@ -1437,7 +1437,7 @@
        if(tracks[i].PT < pt_second_track) continue; // ptcut on tracks
        float genDeltaR = DeltaR(part.Phi(),part.Eta(),tracks[i].Phi,tracks[i].Eta); 
          if(
-             (genDeltaR > deltar) ||
+             //(genDeltaR > deltar) ||
              (genDeltaR==0)  // rejets the track of the particle itself
            ) continue ;
            deltar=genDeltaR;	// finds the closest track
