diff -urN Delphes_V_2.0.1/interface/PdgParticle.h Delphes_V_2.0.1-memleak/interface/PdgParticle.h
--- Delphes_V_2.0.1/interface/PdgParticle.h	2012-02-02 00:10:42.000000000 +0900
+++ Delphes_V_2.0.1-memleak/interface/PdgParticle.h	2012-04-25 18:15:34.769727182 +0900
@@ -44,7 +44,7 @@
 	PdgParticle(const long int pid, const std::string& name, const float m, const float q, const float gamma, const float ctau);
 	PdgParticle(const PdgParticle& p);
 	PdgParticle& operator=(const PdgParticle& p);
-	~PdgParticle() {};
+	virtual ~PdgParticle() {};
 	long int pid() const {return _pid;};
 	float mass() const {return _mass;};
 	float charge() const {return _charge;};
@@ -68,7 +68,7 @@
 	PdgTable(){};
 	PdgTable(const PdgTable& table);
 	PdgTable& operator=(const PdgTable& table);
-        ~PdgTable(){};
+    virtual ~PdgTable(){};
 	void insert(const long int pid, const PdgParticle &p);
 	void print() const;
         PdgParticle operator[](const int pid) const;
diff -urN Delphes_V_2.0.1/src/HepMCConverter.cc Delphes_V_2.0.1-memleak/src/HepMCConverter.cc
--- Delphes_V_2.0.1/src/HepMCConverter.cc	2012-02-02 00:10:40.000000000 +0900
+++ Delphes_V_2.0.1-memleak/src/HepMCConverter.cc	2012-04-25 19:09:26.523451757 +0900
@@ -58,6 +58,8 @@
 {
 
  unsigned int particle_counter=0;
+ index_to_particle.clear();
+ particle_to_index.clear();
  index_to_particle.reserve(evt->particles_size()); 
  index_to_particle[0] = 0; 
  HepMC::GenEvent::vertex_const_iterator v;
