diff -urN Delphes_V_2.0.2/genMakefile.tcl Delphes_V_2.0.2-systemlibs/genMakefile.tcl
--- Delphes_V_2.0.2/genMakefile.tcl	2012-04-30 16:44:59.000000000 +0900
+++ Delphes_V_2.0.2-systemlibs/genMakefile.tcl	2012-09-21 17:26:56.022226759 +0900
@@ -229,9 +229,9 @@
 
 SrcSuf = cc
 
-CXXFLAGS += $(ROOTCFLAGS) -DDROP_CGAL -I. -Iinterface -IUtilities/mcfio -IUtilities/stdhep -IUtilities/Hector/include -IUtilities/CDFCones/interface -IExamples -IUtilities/frog -IUtilities/ExRootAnalysis/interface -IUtilities/Fastjet/include/fastjet -IUtilities/Fastjet/plugins/CDFCones -IUtilities/Fastjet/plugins/CDFCones/interface -IUtilities/Fastjet/plugins/SISCone -IUtilities/CLHEP/Units -IUtilities/HepMC/interface
+CXXFLAGS += $(ROOTCFLAGS) -DDROP_CGAL -I. -Iinterface -IUtilities/mcfio -IUtilities/stdhep -IUtilities/Hector/include -IExamples -IUtilities/frog -IUtilities/ExRootAnalysis/interface -IUtilities/Fastjet/plugins/CDFCones/interface
 # -pg -g
-LIBS = $(ROOTLIBS) -lEG $(SYSLIBS)
+LIBS = $(ROOTLIBS) -lEG $(SYSLIBS) $(shell fastjet-config --libs --plugins) $(shell clhep-config --libs) -lHepMC
 GLIBS = $(ROOTGLIBS) $(SYSLIBS)
 
 ###
@@ -251,7 +251,7 @@
 
 dictDeps {DICT} {Utilities/ExRootAnalysis/src/*LinkDef.h} {src/*LinkDef.h} {Examples/src/*LinkDef.h}
 
-sourceDeps {SOURCE} {src/*.cc} {Utilities/ExRootAnalysis/src/*.cc} {Utilities/Hector/src/*.cc} {Utilities/CDFCones/src/*cc} {Utilities/Fastjet/src/*.cc} {Utilities/Fastjet/plugins/CDFCones/*.cc} {Utilities/Fastjet/plugins/CDFCones/src/*.cc} {Utilities/Fastjet/plugins/SISCone/*.cc} {Utilities/Fastjet/plugins/SISCone/src/*.cc} {Examples/src/*.cc} {Utilities/HepMC/src/*.cc}
+sourceDeps {SOURCE} {src/*.cc} {Utilities/ExRootAnalysis/src/*.cc} {Utilities/Hector/src/*.cc} {Examples/src/*.cc}
 
 stdhepDeps
 
diff -urN Delphes_V_2.0.2/interface/HepMCConverter.h Delphes_V_2.0.2-systemlibs/interface/HepMCConverter.h
--- Delphes_V_2.0.2/interface/HepMCConverter.h	2012-04-30 16:40:29.000000000 +0900
+++ Delphes_V_2.0.2-systemlibs/interface/HepMCConverter.h	2012-07-22 23:58:27.701524854 +0900
@@ -37,10 +37,10 @@
 #include <string>
 #include "DataConverter.h"
 
-#include "GenParticle.h"
-#include "GenVertex.h"
-#include "GenEvent.h"
-#include "IO_AsciiParticles.h"
+#include "HepMC/GenParticle.h"
+#include "HepMC/GenVertex.h"
+#include "HepMC/GenEvent.h"
+#include "HepMC/IO_AsciiParticles.h"
 
 using namespace std;
 
diff -urN Delphes_V_2.0.2/interface/JetsUtil.h Delphes_V_2.0.2-systemlibs/interface/JetsUtil.h
--- Delphes_V_2.0.2/interface/JetsUtil.h	2012-04-30 16:40:27.000000000 +0900
+++ Delphes_V_2.0.2-systemlibs/interface/JetsUtil.h	2012-09-21 16:47:35.056162861 +0900
@@ -50,19 +50,19 @@
 #include <fstream>
 #include <iomanip>
 #include <cstring>
-#include "PseudoJet.hh"
-#include "ClusterSequence.hh"
+#include "fastjet/PseudoJet.hh"
+#include "fastjet/ClusterSequence.hh"
 
 // get info on how fastjet was configured
-#include "Utilities/Fastjet/include/fastjet/config.h"
+#include "fastjet/config.h"
 
 // make sure we have what is needed
-#ifdef ENABLE_PLUGIN_SISCONE
-#  include "SISConePlugin.hh"
+#ifdef FASTJET_ENABLE_PLUGIN_SISCONE
+#  include "fastjet/SISConePlugin.hh"
 #endif
-#ifdef ENABLE_PLUGIN_CDFCONES
-#  include "CDFMidPointPlugin.hh"
-#  include "CDFJetCluPlugin.hh"
+#ifdef FASTJET_ENABLE_PLUGIN_CDFCONES
+#  include "fastjet/CDFMidPointPlugin.hh"
+#  include "fastjet/CDFJetCluPlugin.hh"
 #endif
 
 using namespace std;
diff -urN Delphes_V_2.0.2/src/BFieldProp.cc Delphes_V_2.0.2-systemlibs/src/BFieldProp.cc
--- Delphes_V_2.0.2/src/BFieldProp.cc	2012-04-30 16:40:20.000000000 +0900
+++ Delphes_V_2.0.2-systemlibs/src/BFieldProp.cc	2012-07-23 00:06:38.937306008 +0900
@@ -33,8 +33,8 @@
 
 #include "BFieldProp.h"
 #include "PdgParticle.h"
-#include "SystemOfUnits.h"
-#include "PhysicalConstants.h"
+#include "CLHEP/Units/SystemOfUnits.h"
+#include "CLHEP/Units/PhysicalConstants.h"
 #include<cmath>
 using namespace std;
 
@@ -111,8 +111,8 @@
   // magnetic field parameters
    R_max = DET->TRACK_radius/100.; //[m]
    z_max = DET->TRACK_length/100.; //[m]
-   B_x = DET->TRACK_bfield_x*tesla;
-   B_y = DET->TRACK_bfield_y*tesla;
+   B_x = DET->TRACK_bfield_x*CLHEP::tesla;
+   B_y = DET->TRACK_bfield_y*CLHEP::tesla;
    B_z = DET->TRACK_bfield_z;
 
    loop_overflow_counter=0;
@@ -137,7 +137,7 @@
   //int q1  = ChargeVal(Part->PID) *eplus; // in units of 'e'
   if(Part->M < -999) { // unitialised!
      PdgParticle pdg_part = DET->PDGtable[Part->PID];
-     q  = pdg_part.charge() *eplus; // in units of 'e'
+     q  = pdg_part.charge() *CLHEP::eplus; // in units of 'e'
      M  = pdg_part.mass(); // GeV/c²
   } else  { q = Part->Charge; M = Part->M; }
 
diff -urN Delphes_V_2.0.2/src/HepMCConverter.cc Delphes_V_2.0.2-systemlibs/src/HepMCConverter.cc
--- Delphes_V_2.0.2/src/HepMCConverter.cc	2012-04-30 16:56:46.000000000 +0900
+++ Delphes_V_2.0.2-systemlibs/src/HepMCConverter.cc	2012-07-22 23:58:41.355685550 +0900
@@ -41,10 +41,10 @@
 #include "ExRootTreeBranch.h"
 #include "HepMCConverter.h"
 
-#include "GenParticle.h"
-#include "GenVertex.h"
-#include "IO_AsciiParticles.h"
-#include "IO_GenEvent.h"
+#include "HepMC/GenParticle.h"
+#include "HepMC/GenVertex.h"
+#include "HepMC/IO_AsciiParticles.h"
+#include "HepMC/IO_GenEvent.h"
 
 //-------------------------------------------------------------------------
 int HepMCConverter::find_in_map( const std::map<HepMC::GenParticle*,int>& m, HepMC::GenParticle *p) const
diff -urN Delphes_V_2.0.2/src/JetsUtil.cc Delphes_V_2.0.2-systemlibs/src/JetsUtil.cc
--- Delphes_V_2.0.2/src/JetsUtil.cc	2012-04-30 16:40:17.000000000 +0900
+++ Delphes_V_2.0.2-systemlibs/src/JetsUtil.cc	2012-09-21 16:48:09.294540408 +0900
@@ -71,7 +71,7 @@
   default:
   case 1: {
     // set up a CDF midpoint jet definition
-#ifdef ENABLE_PLUGIN_CDFCONES
+#ifdef FASTJET_ENABLE_PLUGIN_CDFCONES
     plugins = new fastjet::CDFJetCluPlugin(DET->JET_seed,DET->JET_coneradius,DET->JET_C_adjacencycut,DET->JET_C_maxiterations,DET->JET_C_iratch,DET->JET_overlap);
     jet_def = fastjet::JetDefinition(plugins);
 #else
@@ -82,7 +82,7 @@
     
   case 2: {
     // set up a CDF midpoint jet definition
-#ifdef ENABLE_PLUGIN_CDFCONES
+#ifdef FASTJET_ENABLE_PLUGIN_CDFCONES
     plugins = new fastjet::CDFMidPointPlugin (DET->JET_seed,DET->JET_coneradius,DET->JET_M_coneareafraction,DET->JET_M_maxpairsize,DET->JET_M_maxiterations,DET->JET_overlap);
     jet_def = fastjet::JetDefinition(plugins);
 #else
@@ -93,7 +93,7 @@
     
   case 3: {
     // set up a siscone jet definition
-#ifdef ENABLE_PLUGIN_SISCONE
+#ifdef FASTJET_ENABLE_PLUGIN_SISCONE
     plugins = new fastjet::SISConePlugin (DET->JET_coneradius,DET->JET_overlap,DET->JET_S_npass, DET->JET_S_protojet_ptmin);
     jet_def = fastjet::JetDefinition(plugins);
 #else
